### ISSUE TITLE
Upgrade to GraalVM 20.3.x

### ISSUE THREAD

**Issue by usommerl** [usommerl/graalnative4s#16]
I have made an attempt to upgrade to GraalVM v20.3.0 on branch [feature/upgrade-to-graal-v20.3.0][branch] but the created image fails at runtime:

```
Exception in thread "main" com.oracle.svm.core.jdk.UnsupportedFeatureError: Invoke with MethodHandle argument could not be reduced to at most a single call or single field access. The method handle must be a compile time constant, e.g., be loaded from a `static final` field. Method that contains the method handle invocation: java.lang.invoke.LambdaForm$MH/1782707390.invoke_MT(Object, Object)va:148)
            at scala.collection.immutable.$colon$colon.<init>(List.scala:623)
            at scala.collection.immutable.List.prependedAll(List.scala:153)
            at scala.collection.IterableOnceOps.toList(IterableOnce.scala:1258)
            at scala.collection.IterableOnceOps.toList$(IterableOnce.scala:1258)
            at scala.collection.AbstractIterable.toList(Iterable.scala:920)
            at cats.effect.internals.IOTracing$.buildFrame(IOTracing.scala:48)
            at cats.effect.internals.IOTracing$.buildCachedFrame(IOTracing.scala:39)
            at cats.effect.internals.IOTracing$.cached(IOTracing.scala:34)
            at cats.effect.IO.flatMap(IO.scala:133)
            at cats.effect.IOLowPriorityInstances$IOEffect.flatMap(IO.scala:886)
            at cats.effect.IOLowPriorityInstances$IOEffect.flatMap(IO.scala:863)
            at cats.FlatMap$Ops.flatMap(FlatMap.scala:229)
            at cats.FlatMap$Ops.flatMap$(FlatMap.scala:229)
            at cats.FlatMap$ToFlatMapOps$$anon$2.flatMap(FlatMap.scala:243)
            at io.odin.loggers.DefaultLogger.$anonfun$log$1(DefaultLogger.scala:23)
            at cats.Applicative.whenA(Applicative.scala:187)
            at cats.Applicative.whenA$(Applicative.scala:186)
            at cats.effect.IOLowPriorityInstances$IOEffect.whenA(IO.scala:863)
            at io.odin.loggers.DefaultLogger.log(DefaultLogger.scala:23)
            at io.odin.loggers.DefaultLogger.$anonfun$info$1(DefaultLogger.scala:98)
            at cats.Applicative.whenA(Applicative.scala:187)
            at cats.Applicative.whenA$(Applicative.scala:186)
            at cats.effect.IOLowPriorityInstances$IOEffect.whenA(IO.scala:863)
            at io.odin.loggers.DefaultLogger.info(DefaultLogger.scala:98)
            at server.Main$.runF(Main.scala:26)
            at server.Main$.run(Main.scala:21)
            at cats.effect.IOApp.$anonfun$main$3(IOApp.scala:68)
            at cats.effect.internals.IOAppPlatform$.mainFiber(IOAppPlatform.scala:40)
            at cats.effect.internals.IOAppPlatform$.main(IOAppPlatform.scala:25)
            at cats.effect.IOApp.main(IOApp.scala:68)
            at cats.effect.IOApp.main$(IOApp.scala:67)
            at server.Main$.main(Main.scala:17)
            at server.Main.main(Main.scala)
```

See [graal#2761][graal-issue]

[branch]: https://github.com/usommerl/graalnative4s/tree/feature/upgrade-to-graal-v20.3.0
[graal-issue]: https://github.com/oracle/graal/issues/2761

**Comment by usommerl** [comment]
Maybe this could be helpful: [scalameta/svm-subs][svm-subs]


[svm-subs]: https://github.com/scalameta/svm-subs

**Comment by usommerl** [comment]
[graal#2928](https://github.com/oracle/graal/issues/2928) is describing the same symptoms

**Comment by usommerl** [comment]
I'm closing this issue because [GraalVM v21.0.0][1] was released and it is unclear whether there will be a fix in v20.3.x for the issue described above.

[1]: https://www.graalvm.org/release-notes/21_0/

**Comment by usommerl** [comment]
Superseded by #51
